pkg/orchestration/wiring/ec2compute/common/common.go

```diff
@@ -139,11 +141,15 @@ func WireUp(microServiceNameInSpec, ec2ComputePlanName string, stateResource *or
 
 	for _, dependency := range dependencies {
 		if !dependency.Contract.IsEmpty() {
```

Comment:
I think this should work differently. It should look for shapes it is interested in and if not found fall back to current behavior. Then once everything exposes a contract, we remove the fallback and return an error instead.

Reply:
I use outputResourceContract to mark resources that are migrated to contracts already, but I am not consistent and use if resourceType = X in the same "if". I should remove "if dependency.Type" check. It is unnecessary.